be/src/exec/aggregator.cpp

```diff
@@ -194,6 +194,11 @@ ChunkUniquePtr AggregatorParams::create_result_chunk(bool is_serialize_fmt, cons
 Aggregator::Aggregator(AggregatorParamsPtr params) : _params(std::move(params)) {}
 
 Status Aggregator::open(RuntimeState* state) {
+    std::lock_guard stage_lock(_agg_run_stage_mutex);
```

Comment:
if we do once, can we use once_flag instead of lock_guard?
A better design should be that adopting JIT is hid and when invoking Expr::open,  Expr checks a JIT cache to see if the Expr already exists in JIT cache, otherwise jit compile the expr and popolate the JIT cache.

Reply:
not really just once. if the first dirver is doing prepare, other dirvers do the following open/next will result errors as the unfinished prepare(). so others should wait prepread() and opend() are done.

JIT already use cache to avoid compiling a replicated exrp with the same name.